pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -361,76 +361,133 @@ function validateParams(vmParams) {
     }
 }
 
-export const createVmDialog = (dispatch, osInfoList) => {
-    const vmParams = {
-        'vmName': '',
-        "sourceType": COCKPIT_FILESYSTEM_SOURCE,
-        'source': '',
-        'vendor': NOT_SPECIFIED,
-        "os": OTHER_OS_SHORT_ID,
-        'memorySize': 1024, // MiB
-        'storageSize': 10, // GiB
-        'startVm': false,
-        'error': null,
-    };
-
-    const changeParams = (key, value) => {
-        vmParams[key] = value;
-    };
-
-    const vendors = prepareVendors(osInfoList);
-
-    const dialogBody = (
-        <CreateVM vmParams={vmParams}
-                  familyList={vendors.familyList}
-                  familyMap={vendors.familyMap}
-                  vendorMap={vendors.vendorMap}
-                  valuesChanged={changeParams} />
-    );
-
-    const dialogProps = {
-        'title': _("Create New Virtual Machine"),
-        'body': dialogBody,
-    };
-
-    // also test modifying properties in subsequent render calls
-    const footerProps = {
-        'actions': [
-            {
-                'clicked': () => {
-                    const error = validateParams(vmParams);
-                    if (error) {
-                        return cockpit.defer().reject(error).promise;
-                    } else {
-                        // leave dialog open to show immediate errors from the backend
-                        // close the dialog after VMS_CONFIG.LeaveCreateVmDialogVisibleAfterSubmit
-                        // then show errors in the notification area
-                        return timeoutedPromise(
-                            dispatch(createVm(vmParams)),
-                            VMS_CONFIG.LeaveCreateVmDialogVisibleAfterSubmit,
-                            null,
-                            (exception) => {
-                                dispatch(addErrorNotification({
-                                    message: cockpit.format(_("Creation of vm $0 failed"), vmParams.vmName),
-                                    description: exception,
-                                }));
-                            });
-                    }
+class CreateVmModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = this.initialState;
+        this.onValueChanged = this.onValueChanged.bind(this);
+        this.onCreateClicked = this.onCreateClicked.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+        this.dialogErrorDismiss = this.dialogErrorDismiss.bind(this);
+    }
+
+    get initialState() {
```

Comment:
OOI, is this some recommended pattern? Usually the constructor should just set the initial state, and be done with it, and this isn't being used anywhere else. The days of forms with a [ Reset ] button are long gone.. :)

Reply:
I didn't see it anywhere, maybe found it cleaner, but since it's not used anywhere else I 'll agree it's not needed. Moving everything to constructor.